src/v/cluster/archival_metadata_stm.cc

```diff
@@ -186,12 +187,11 @@ command_batch_builder archival_metadata_stm::batch_start(
     return {*this, deadline, as};
 }
 
-std::vector<archival_metadata_stm::segment>
+fragmented_vector<archival_metadata_stm::segment>
 archival_metadata_stm::segments_from_manifest(
   const cloud_storage::partition_manifest& manifest) {
-    std::vector<segment> segments;
-    segments.reserve(manifest.size());
-
+    fragmented_vector<segment> segments;
+    // TODO segments.reserve(manifest.size())
```

Comment:
IIRC the reason fragmented vector doesn't have a `reserve()` method is just that it wouldn't do anything different than a bunch of pushes (because it always allocates whole fragment-sized extents).

In that case we don't need these TODO lines about reserving

Reply:
I kept it as a reminder when fragment_vector will have a preallocation option to preallocate all the fragments, but i can remove the comment